### PR TITLE
Update smoke test Java versions

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: smoke-tests/apps/CoreAndFilter3xUsingOld3xAgent/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CoreAndFilter3xUsingOld3xAgentTest.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_11;
 import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_11_OPENJ9;
 import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_17;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_17_OPENJ9;
```
Comment (harsimar) [inline smoke-tests/apps/CoreAndFilter3xUsingOld3xAgent/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CoreAndFilter3xUsingOld3xAgentTest.java:9]: is java 21 not considered "old" for this test? just curious

Comment (trask) [reply smoke-tests/apps/CoreAndFilter3xUsingOld3xAgent/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CoreAndFilter3xUsingOld3xAgentTest.java:9]: the issue is that the old 3.x agent (which this is a weird test of) doesn't support Java 21, there's a related comment below:

>  // note: old 3.x agents don't support Java 19

File: smoke-tests/apps/JmxMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/JmxMetricTest.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_11;
 import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_17;
 import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_21;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_23;
```
Comment (heyams) [inline smoke-tests/apps/JmxMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/JmxMetricTest.java:9]: why is this missing JAVA_23_OPENJ9?

Comment (trask) [reply smoke-tests/apps/JmxMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/JmxMetricTest.java:9]: this test has never run on (any versions of) OpenJ9

I suspect it's due to OpenJ9 exposing different JMX metrics, but looking into it now, and will at least add comment in this file explaining...

Comment (trask) [reply smoke-tests/apps/JmxMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/JmxMetricTest.java:9]: comment added (below)

**Review by jeanbisutti — APPROVED** [review]

**Pull request merged**